app/services/resource_recommendation.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+class ResourceRecommendation
+  def initialize(moment)
+    @moment = moment
+    @moment_keywords = []
+    @matched_resources = []
+  end
+
+  def get_keywords(array)
+    array.each do |item|
+      @moment_keywords.push(item['name'].split,
+                            html_clean(item['description']).split)
+    end
+  end
+
+  def modify_keywords
+    @moment_keywords = @moment_keywords.flatten
+    @moment_keywords.each do |keyword|
+      keyword.gsub!(%r{([_@#!%()\-=;><,{}\~\[\]\./\?\"\*\^\$\+\-]+)}, '')
+    end
+    @moment_keywords = @moment_keywords.map(&:downcase)
+  end
+
+  def match_keywords
+    all_resources.each do |resource|
+      resource_tags = resource['tags'].map do |tag|
+        tag.split('_')
+      end
+      unless (resource_tags.flatten & @moment_keywords).empty?
+        @matched_resources.push(resource)
+      end
+    end
+  end
+
+  def resources
+    get_keywords(@moment.categories)
+    get_keywords(@moment.moods)
+    get_keywords(@moment.strategies)
+    @moment_keywords.push(moment_name, moment_why, moment_fix)
+    modify_keywords
```

Comment:
You can move all methods related to extracting keywords from a moment to a separated class. That way the extracting step will be separated from matching. This will make the codebase easier to test and maintain.